CMakeLists.txt

```diff
@@ -136,18 +151,27 @@ if (USE_SIMD AND USE_SIMD_NEON)
   target_compile_definitions(PFFFT PRIVATE PFFFT_ENABLE_NEON=1)
   target_compile_options(PFFFT PRIVATE "-mfpu=neon")
 endif()
-if (USE_SIMD AND USE_TYPE_DOUBLE)
-  if(WIN32)
-    if(DISABLE_SIMD_AVX)
-      set_property(SOURCE pffft_double.c PROPERTY COMPILE_FLAGS "/arch:SSE2")
-    else()
-      set_property(SOURCE pffft_double.c PROPERTY COMPILE_FLAGS "/arch:AVX")
-    endif()
+if (USE_SIMD AND WIN32 AND MINGW)
+  message(STATUS "Setting PFFFT target architecture for MINGW")
+  if(DISABLE_SIMD_AVX)
+    target_compile_options(PFFFT PRIVATE "-march=core2")
   else()
-    set_property(SOURCE pffft_double.c PROPERTY COMPILE_FLAGS "-march=native")
+    target_compile_options(PFFFT PRIVATE "-march=sandybridge")
   endif()
-  if(DISABLE_SIMD_AVX)
-    target_compile_definitions(PFFFT PRIVATE PFFFT_AVX_DISABLE=1)
+else()
+  if (USE_SIMD AND USE_TYPE_DOUBLE)
+    if(WIN32)
+      if(DISABLE_SIMD_AVX)
+        set_property(SOURCE pffft_double.c PROPERTY COMPILE_FLAGS "/arch:SSE2")
+      else()
+        set_property(SOURCE pffft_double.c PROPERTY COMPILE_FLAGS "/arch:AVX")
+      endif()
+    else()
+      set_property(SOURCE pffft_double.c PROPERTY COMPILE_FLAGS "-march=native")
```

Comment:
Why not `-march=core2`?

Reply:
that line with `-march=native` already existed, that i didn't think much about it.
could think it should depend on a cmake option like "optimize for local computer? (or minimum required, e.g. for distribution)". ideas on this?
core2: intrinsics for double could be using avx, which was '-march=sandybridge' as used for mingw cross compilation.
have the target a text/string cmake option?

Reply:
I saw that you changed `-march=native` further above to `-march=core2`, that's why I asked. In general using `-march=native` is not a good idea for anyone who intends to package the build result -- your users might use a worse architecture than the machine you're building on.

It seems there is no built in option in CMake, as implied by [this SO question]( https://stackoverflow.com/questions/46724267/cmake-march-hardware).

Reply:
in a [commit](https://github.com/dievsky/pffft/commit/d90577e0f260199a8f1f43835674a4628795443f) on the unaligned experiments branch by @dievsky, there is this:
    
    if(DISABLE_SIMD_AVX)
      set_property(SOURCE pffft_double.c PROPERTY COMPILE_FLAGS "-msse2")
    else()
      set_property(SOURCE pffft_double.c PROPERTY COMPILE_FLAGS "-mavx")

i seem to remember that it was working with gcc on my debian machine, but i'm not sure now.